lib/cocoapods/installer/xcode/target_uuid_generator.rb

```diff
@@ -0,0 +1,31 @@
+module Pod
+  class Installer
+    # Generates stable UUIDs for Native Targets.
+    #
+    class TargetUUIDGenerator < Xcodeproj::Project::UUIDGenerator
+      # This method override is used to ONLY generate stable UUIDs for PBXNativeTarget instances and no other type.
+      # Stable native target UUIDs are necessary for incremental installation because other projects reference the
+      # target by its UUID in the remoteGlobalIDString field.
+      #
+      # @param [Array<Project>] projects
+      #        The list of projects used to generate stabe target UUIDs.
+      #
+      def generate_all_paths_by_objects(projects)
+        @paths_by_object = {}
+        all_objects = projects.flat_map(&:objects)
+        all_objects.each do |object|
+          @paths_by_object[object] = if object.is_a? Xcodeproj::Project::Object::AbstractTarget
+                                       project_basename = object.project.path.basename.to_s
```

Comment:
probably worth pre-computing this on the project, or maybe instead of flat-mapping to the objects, loop though each project, so we don't need to do this for every single object?